### PR TITLE
Better generated Curl example

### PR DESCRIPTION
**Pull request by jirikuncar**
### What does this PR do?

Uses Hugo to generate Curl examples from OpenAPI spec.

### Motivation

Respect selected region in curl example.

### Preview link

https://docs-staging.datadoghq.com/jirikuncar/curl-preview/api/

### CODE REVIEW

File: layouts/api/list.html [hunk 1]
```diff
@@ -29,11 +29,11 @@
         {{ if eq $region "com"}}
             {{ $region = "us"}}
         {{ end }}
-
```
Comment (jirikuncar) [inline layouts/api/list.html:32]: should I revert it?

File: layouts/api/list.html [hunk 2]
```diff
@@ -368,37 +369,82 @@ <h3 class="mb-2">Code Example</h3>
...
+{{- with $pathParams -}}
+{{- range . -}}
+export {{ .name }}="{{ .example | default "CHANGE_ME"}}"
+{{ end -}}
+{{- end -}}
+{{ if $context.action.requestBody -}}
+(cat << EOF
+{{ ($.Scratch.Get "jsonRequestBody") }}
+EOF
+) | curl -X <span class="text-uppercase">{{ $context.actionType }}</span> \
```
Comment (gzussa) [inline layouts/api/list.html:418]: Let's have the payload after the command. In a more traditional way.

File: layouts/api/list.html [hunk 3]
```diff
@@ -368,37 +369,82 @@ <h3 class="mb-2">Code Example</h3>
...
+                                {{ $count := 1 }}
                                 {{ range . }}
 
                                     {{ $file_content := .Content }}
                                     {{ $file_extension := index (split (path.Ext .) ".") 1 }}
                                     {{ $lang := index $dot.Site.Data.highlighting.languages $file_extension }}
-                            
+
+                                    {{ if eq $file_extension "sh"}}
+                                    {{ else }}
```
Comment (davidejones) [inline layouts/api/list.html:435]: You can probably change these if there is no logic for here
```suggestion
                                    {{ if ne $file_extension "sh"}}
```

File: layouts/api/list.html [hunk 4]
```diff
@@ -368,37 +369,82 @@ <h3 class="mb-2">Code Example</h3>
...
+                                    </li>
+
+                                    {{ $count := 1}}
                                     {{ range . }}
 
                                     {{ $file_extension := index (split (path.Ext .) ".") 1 }}
                                     {{ $lang := index $dot.Site.Data.highlighting.languages $file_extension }}
 
+                                    {{ if eq $file_extension "sh"}}
+                                    {{ else }}
```
Comment (davidejones) [inline layouts/api/list.html:386]:
```suggestion
                                    {{ if ne $file_extension "sh"}}
```

File: layouts/api/list.html (hunk 1 above)
Comment (davidejones) [reply layouts/api/list.html:32]: should be fine

File: layouts/api/list.html [hunk 5]
```diff
@@ -368,37 +369,87 @@ <h3 class="mb-2">Code Example</h3>
...
+                                {{ $lang := index $dot.Site.Data.highlighting.languages "sh" }}
+                                <div class="code-snippet js-code-block {{if eq $file_extension "sh"}} default {{end }} {{ if eq $count 0}}d-block{{else}}d-none{{end}} code-block-{{ $lang.pygments_name }}" >
+                                    <div class="code-button-wrapper position-absolute">
+                                        <button class="btn text-primary js-copy-button">Copy</button>
+                                    </div>
+                                    {{ $pathParams := $.Scratch.Get "pathParams" }}
+                                    <div class="highlight">
+                                        <pre class="chroma">
+                                            <code class="language-fallback" data-lang="fallback">
+{{- with $pathParams -}}
```
Comment (zbayoff) [inline layouts/api/list.html:408]: Can you indent this block below to follow the rest of the code style?

Comment (jirikuncar) [reply layouts/api/list.html:408]: I am not sure how to make the inside code not indented.

**Pull request merged**